LIBITD-2541. Implemented the containment check for an IP address in a…

mackenziekilgore opened the pull request:
… group

Created some tests, but will work on more.
Added cidr as a submodule in src/ipmanager/api.
Updated the views to work with the contains method

https://umd-dit.atlassian.net/browse/LIBITD-2541

Commit:
LIBITD-2541. Implemented the containment check for an IP address in a group

Created some tests, but will work on more.
Added cidr as a submodule in src/ipmanager/api.
Updated the views to work with the contains method

https://umd-dit.atlassian.net/browse/LIBITD-2541

peichman-umd reviewed (changes requested):
This commit includes not just reference to the cidr library as a submodule, but also copies of the actual files, which is not what we want.

Suggested steps:
* Remove the `src/ipmanager/api/cidr` directory
* Change the submodule path in `.gitmodules` to `src/cidr`
* Update the `import` statement in the `views.py` to be `from cidr.cidr import Cidr, CidrSet` (no leading `.`)

Commit:
LIBITD-2541. Made changes to not include copies of actual cidr files

Left the . before cidr.cidr because it wasn't working.

https://umd-dit.atlassian.net/browse/LIBITD-2541

peichman-umd commented:
The cidr source files are still included in the PR. You will need to remove them using `git rm -rf src/ipmanager/api/cidr`, and commit those changes.

Commit:
LIBITD-2541. Removed cidr source files

https://umd-dit.atlassian.net/browse/LIBITD-2541

peichman-umd commented:
One more thing: You need to add the submodule reference at `src/cidr` using `git add src/cidr`. Then, when you run `git diff --cached` you should see something like this:

```diff
diff --git a/src/cidr b/src/cidr
new file mode 160000
index 0000000..a91a1b4
--- /dev/null
+++ b/src/cidr
@@ -0,0 +1 @@
+Subproject commit a91a1b47a8de3ac00d8b59b0b6117c8613623824
```

(Submodules are tricky :-)

Commit:
LIBITD-2541. Added the submodule reference

Also deleted the . before cidr.cidr

https://umd-dit.atlassian.net/browse/LIBITD-2541

Commit:
LIBITD-2541. Added the models file for the commit.

I forgot to add it in the last commit.

https://umd-dit.atlassian.net/browse/LIBITD-2541
